src/basic/cgroup-util.c

```diff
@@ -267,7 +267,8 @@ int cg_kill(
                                 continue;
 
                         if (log_kill)
-                                log_kill(pid, sig, userdata);
+                                if (log_kill(pid, sig, userdata))
+                                        return 1;
```

Comment:
a few lines further down there's currently this logic:

```c
if (ret == 0)
        ret = 1
```

i.e. this sets the return value to > 0 as soon as at least one process was killed. I think this should be conditionalized depending on whether log_kill is passed. i.e. the return value is determine by log_kill if it is set, and otherwise the return value is > 0 if at least one process was killed, as before. i.e. change that code to a simple:

```c
if (!log_kill && ret == 0)
        ret = 1;
```